File: app/src/main/java/org/mifos/mobile/ui/fragments/RegistrationVerificationFragment.java
package org.mifos.mobile.ui.fragments;

import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.Toast;

import org.mifos.mobile.R;
import org.mifos.mobile.models.register.UserVerify;
import org.mifos.mobile.presenters.RegistrationVerificationPresenter;
import org.mifos.mobile.ui.activities.LoginActivity;
import org.mifos.mobile.ui.activities.base.BaseActivity;
import org.mifos.mobile.ui.fragments.base.BaseFragment;
import org.mifos.mobile.ui.views.RegistrationVerificationView;
import org.mifos.mobile.utils.Toaster;

import javax.inject.Inject;

import androidx.annotation.Nullable;
import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

/**
 * Created by dilpreet on 31/7/17.
 */

public class RegistrationVerificationFragment extends BaseFragment implements
        RegistrationVerificationView {

    @BindView(R.id.et_request_id)
    EditText etRequestId;

    @BindView(R.id.et_authentication_token)
    EditText etToken;

    @Inject
    RegistrationVerificationPresenter presenter;

    private View rootView;

    public static RegistrationVerificationFragment newInstance() {
        RegistrationVerificationFragment fragment = new RegistrationVerificationFragment();
        return fragment;
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
            @Nullable Bundle savedInstanceState) {
        rootView = inflater.inflate(R.layout.fragment_registration_verification, container, false);

        ((BaseActivity) getActivity()).getActivityComponent().inject(this);
        ButterKnife.bind(this, rootView);
        presenter.attachView(this);

        return rootView;
    }

    @OnClick(R.id.btn_verify)
    public void verifyClicked() {
        UserVerify userVerify = new UserVerify();
        userVerify.setAuthenticationToken(etToken.getText().toString());
        userVerify.setRequestId(etRequestId.getText().toString());
        presenter.verifyUser(userVerify);
    }

    @Override
    public void showVerifiedSuccessfully() {
        startActivity(new Intent(getActivity(), LoginActivity.class));
        Toast.makeText(getContext(), getString(R.string.verified), Toast.LENGTH_SHORT).show();
        getActivity().finish();
    }

    @Override
    public void showError(String msg) {
        Toaster.show(rootView, msg);
    }

    @Override
    public void showProgress() {
        showMifosProgressDialog(getString(R.string.verifying));
    }

    @Override
    public void hideProgress() {
        hideMifosProgressDialog();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        presenter.detachView();
    }
}

File: app/src/main/java/org/mifos/mobile/ui/fragments/RegistrationVerificationFragment.kt
package org.mifos.mobile.ui.fragments

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.lifecycle.ViewModelProvider
import dagger.hilt.android.AndroidEntryPoint
import org.mifos.mobile.R
import org.mifos.mobile.databinding.FragmentRegistrationVerificationBinding
import org.mifos.mobile.ui.activities.LoginActivity
import org.mifos.mobile.ui.fragments.base.BaseFragment
import org.mifos.mobile.utils.RegistrationUiState
import org.mifos.mobile.utils.Toaster
import org.mifos.mobile.viewModels.RegistrationViewModel

/**
 * Created by dilpreet on 31/7/17.
 */
@AndroidEntryPoint
class RegistrationVerificationFragment : BaseFragment() {
    private var _binding: FragmentRegistrationVerificationBinding? = null
    private val binding get() = _binding!!
    private lateinit var viewModel: RegistrationViewModel

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?,
    ): View {
        _binding = FragmentRegistrationVerificationBinding.inflate(inflater, container, false)
        val rootView = binding.root
        viewModel = ViewModelProvider(this)[RegistrationViewModel::class.java]
        return rootView
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        viewModel.registrationVerificationUiState.observe(viewLifecycleOwner) { state ->
            when (state) {
                RegistrationUiState.Loading -> showProgress()

                RegistrationUiState.Success -> {
                    hideProgress()
                    showVerifiedSuccessfully()
                }

                is RegistrationUiState.Error -> {
                    hideProgress()
                    showError(getString(state.exception))
                }
            }
        }

        binding.btnVerify.setOnClickListener {
            verifyClicked()
        }
    }

    private fun verifyClicked() {
        val authenticationToken = binding.etAuthenticationToken.text.toString()
        val requestId = binding.etRequestId.text.toString()
        viewModel.verifyUser(authenticationToken, requestId)
    }

    private fun showVerifiedSuccessfully() {
        startActivity(Intent(activity, LoginActivity::class.java))
        Toast.makeText(context, getString(R.string.verified), Toast.LENGTH_SHORT).show()
        activity?.finish()
    }

    fun showError(msg: String?) {
        Toaster.show(binding.root, msg)
    }

    fun showProgress() {
        showMifosProgressDialog(getString(R.string.verifying))
    }

    fun hideProgress() {
        hideMifosProgressDialog()
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }

    companion object {
        fun newInstance(): RegistrationVerificationFragment {
            return RegistrationVerificationFragment()
        }
    }
}
